Allow full adjustment of srsRAN configuration through Helm values.yaml parameters

bradunov opened the pull request:
Currently, the srsRAN config parameters are hardcoded in a config file that is stored in the Docker image. Only a handful of parameters can be changed from the Helm values. This PR allows users to pass a full srsRAN config file, adding it under a new key `srs_ran_config`. A desired srsRAN configuration can simply be copied under this key and it will be taken as the template. The Helm install process will still adjust some parameters further, as before. This allows user to easily adjust some parts of the config that change more often.

bradunov committed:
Add default srsRAN config to Helm chart values.yaml

ujjwalpawar commented:
This would solve one of our issue. But another variable is needed in value.yaml for nodeselector to make sure the jrtc and RAN are deployed on same node.  This is an example of  what i have added. 
**In values.yaml** 
```yaml
nodeSelector:
  hostname: k8s-worker6
```
**In templates/gnb-statefulset.yaml** 
```yaml
....
    spec:
      securityContext:
        runAsUser: 0
        runAsGroup: 0
      serviceAccountName: srs-service-account
      nodeSelector:
        kubernetes.io/hostname: {{ .Values.nodeSelector.hostname }}
      terminationGracePeriodSeconds: 0
      volumes:
      - name: dshm
....
```
**In templates/jrtc.yaml**
```yaml
....
    spec:
      securityContext:
        runAsUser: 0
        runAsGroup: 0
      serviceAccountName: srs-service-account
      terminationGracePeriodSeconds: 0
      nodeSelector:
        kubernetes.io/hostname: {{ .Values.nodeSelector.hostname }}
      volumes:
      - name: dshm
....
````

bradunov committed:
Fix typo

bradunov commented:
> This would solve one of our issue. But another variable is needed in value.yaml for nodeselector to make sure the jrtc and RAN are deployed on same node. This is an example of what i have added. **In values.yaml**
> 
> ```yaml
> nodeSelector:
>   hostname: k8s-worker6
> ```
> 
> **In templates/gnb-statefulset.yaml**
> 
> ```yaml
> ....
>     spec:
>       securityContext:
>         runAsUser: 0
>         runAsGroup: 0
>       serviceAccountName: srs-service-account
>       nodeSelector:
>         kubernetes.io/hostname: {{ .Values.nodeSelector.hostname }}
>       terminationGracePeriodSeconds: 0
>       volumes:
>       - name: dshm
> ....
> ```
> 
> **In templates/jrtc.yaml**
> 
> ```yaml
> ....
>     spec:
>       securityContext:
>         runAsUser: 0
>         runAsGroup: 0
>       serviceAccountName: srs-service-account
>       terminationGracePeriodSeconds: 0
>       nodeSelector:
>         kubernetes.io/hostname: {{ .Values.nodeSelector.hostname }}
>       volumes:
>       - name: dshm
> ....
> ```

Since you now added this in a separate PR, are you happy with me closing it?

matthewbalkwill committed:
Merge branch 'main' into bozidar/srs_config

matthewbalkwill committed:
Update values to latest srsran default config, and remove msft_config.yaml

matthewbalkwill committed:
Add instructions for building 'zmq' and 'srsue' images

matthewbalkwill committed:
Do not use 'default_config.yaml' in ZMQ mode

matthewbalkwill committed:
Add note explaining that 'srs_ran_config' is not used in 'zmq' mode
